### PR TITLE
Zarmada/working/ADTS-160

### PR DESCRIPTION
**Pull request by miltonzarmada**
### Testing 🧪
 > [ Add any special instructions needed to test or view your changes such as environment URLs or other config details. ]

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: src/Components/OATGraphViewer/Internal/OATGraphCustomNode.tsx [hunk 1]
```diff
@@ -43,6 +50,13 @@ const OATGraphCustomNode: React.FC<IOATGraphCustomNodeProps> = ({
     const actionButtonStyles = getGraphViewerActionButtonStyles();
 
     const onNameChange = (evt) => {
+        if (evt.target.value.length <= OATDisplayNameLengthLimit) {
+            setNameText(evt.target.value.length);
```
Comment (jamescarpinter) [inline src/Components/OATGraphViewer/Internal/OATGraphCustomNode.tsx:54]: Why set this to length?

File: src/Components/OATGraphViewer/Internal/OATGraphCustomNode.tsx [hunk 2]
```diff
@@ -100,7 +114,14 @@ const OATGraphCustomNode: React.FC<IOATGraphCustomNodeProps> = ({
...
     const onIdChange = (evt) => {
-        setIdText(evt.target.value);
+        if (evt.target.value.length <= OATIdLengthLimit) {
+            setIdText(evt.target.value);
+            setIdError(null);
+        } else {
+            setIdError(true);
+        }
+
+        setNameText(evt.target.value);
```
Comment (jamescarpinter) [inline src/Components/OATGraphViewer/Internal/OATGraphCustomNode.tsx:124]: Why do we set name on an ID change?

File: src/Components/OATPropertyEditor/FormAddEnumItem.tsx [hunk 3]
```diff
@@ -191,6 +200,20 @@ export const FormAddEnumItem = ({
         return find ? `${t('OATPropertyEditor.errorRepeatedEnumValue')}` : '';
     };
 
+    const handleNameChange = (value) => {
+        const regularExpression = new RegExp(
+            '^[a-zA-Z](?:[a-zA-Z0-9_]*[a-zA-Z0-9])?$'
```
Comment (jamescarpinter) [inline src/Components/OATPropertyEditor/FormAddEnumItem.tsx:205]: I'd put a short explainer as a comment here so folks don't have to try and parse in their heads.

File: src/Components/OATPropertyEditor/FormAddEnumItem.tsx [hunk 4]
```diff
@@ -546,16 +611,27 @@ export const FormAddEnumItem = ({
...
                 />
             </div>
 
             <PrimaryButton
                 text={t('OATPropertyEditor.update')}
                 allowDisabledFocus
                 onClick={handleAddEnumValue}
-                disabled={error || !enumValue || !name}
+                disabled={
+                    error ||
```
Comment (jamescarpinter) [inline src/Components/OATPropertyEditor/FormAddEnumItem.tsx:627]: We might need to rename error now that we have multiple types?

File: src/Models/Constants/Constants.ts [hunk 5]
```diff
@@ -194,6 +194,10 @@ export const OATRelationshipHandleName = 'Relationship';
 export const OATComponentHandleName = 'Component';
 export const OATExtendHandleName = 'Extend';
 export const OATInterfaceType = 'Interface';
+export const OATCommentLengthLimit = 512;
+export const OATDescriptionLengthLimit = 512;
+export const OATDisplayNameLengthLimit = 64;
+export const OATIdLengthLimit = 2048;
```
Comment (jamescarpinter) [inline src/Models/Constants/Constants.ts:200]: This file might be a good place for the regex string too?

**Review by jamescarpinter — APPROVED** [review]

**Comment by jamescarpinter** [comment]
@miltonzarmada I can't see the definition of `DTDLNameRegex` in the constants file?

**Comment by miltonzarmada** [comment]
The definition of DTDLNameRegex should be under src>models>constants>constants.ts

Looks like it was already part of the code base
@jamescarpinter

**Pull request merged**